### DISCUSSION TITLE
Pump goes off randomly.

### DISCUSSION THREAD

**Discussion by HankM34** [visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA#264] [General]
Hi all,

I have Lay-z-spa Miami about 4 years old. Pump model is 54154 eggshape, 4 wire version. I have soldered nodemcu + logic signal converter to the prototype board. I have noticed two problems,

1) when I am using panel to control the unit (with module fitted), after few minutes it switching off with error code E02, it is not doing it without module.
2) when the pump is switched on from webpanel then the pump goes off randomly and back on, also the device is sometimes hanging and is not accessible. 

Regards,

Wojciech

**Comment by chunkysteveo** [comment]
Sounds similar to when I first started with my 4-wire version. I had weird start-stop issue with the pump and random errors.

Turns out for me it was (or may have been) the connections for the cables to the PCB. I used dupoint connectors with pin headers that could be removed easily. I removed those and just soldered the two cables directly to the PCB - never EVER had an issue with the pump stopping and starting or errors again after that. There must have been a loose connection, or dry solder joint - most likely the cheap dupoint connectors and header pins causing an intermittent/poor connection.

Worth testing to eliminating if your cables are not directly soldered?

**Comment by visualapproach** [comment]
Hi,
also worth checking: wifi strength, shorter wires.

**Comment by HankM34** [comment]
Hi all,

Yesterday I have soldered new board, please see attached pictures. 
![IMG_5388](https://user-images.githubusercontent.com/42208475/167657952-74de08c5-3fd1-450b-b44c-4ea4d8d643f6.jpg)
![IMG_5389](https://user-images.githubusercontent.com/42208475/167657986-d2950d45-51b9-4f4a-99f7-567944dbed92.jpg)
Cables soldered to the board directly.

Unfortunately same issue, it is a bit better when the board is out of pump but still randomly goes off or E02. 
Wifi access point is next to the pump, so it may to do not be the signal issue.
Today I will try to flash nodemcu with blank image and flash again with WiFi-remote-for-Bestway-Lay-Z-SPA.

Regards,

Wojciech

**Reply by visualapproach** [reply]
If you check last reboot reason on the config page and it says hardware watchdog, you may need to lower the websockets timeout setting. 
If it restarts due to power on or something you might solve it by putting a 100-1000 uF cap over 5V-GND.
And you may wanna cut the (blue/red/green/black) wires to be as short as possible. But keep a few cm to be able to reuse the connectors if needed.
You can also try the other type of level converter. Some people reports success when switching, even if it's on the new 6-wire systems.

**Comment by HankM34** [comment]
Hi,

I have soldered 1000uF/16v capacitor, short connector cables and blanked flash. Unfortunately same issue:
After power up, filter pump switched on from panel, after few minutes take control from web panel and pump off, immediately error E02. Today I will try to order new level converter.

Thanks
